pyquantifier/data.py

```diff
@@ -327,6 +328,22 @@ def generate_calibration_curve(self, method='platt scaling', num_bin=10):
             return prob_cali_obj
         elif method == 'temperature scaling':
             pass
+        elif method == 'isotonic regression':
+            train_CX = self.df['p_pos'].values
+            train_GT = self.df['gt_label'].map({'neg': 0, 'pos': 1}).values
+
+            # Sort the train_CX and train_GT arrays
+            sorted_indices = np.argsort(train_CX)
+            sorted_train_CX = train_CX[sorted_indices]
+            sorted_train_GT = train_GT[sorted_indices]
+
+            # Fit isotonic regression model
+            ir = IsotonicRegression(out_of_bounds='clip')
+            y_axis = ir.fit_transform(sorted_train_CX, sorted_train_GT)
```

Comment:
I think this should not be `.fit_transform()`

Based on the [doc](https://scikit-learn.org/dev/modules/generated/sklearn.isotonic.IsotonicRegression.html#sklearn.isotonic.IsotonicRegression.fit_transform) of Isotonic Regression, 
> Fits transformer to X and y with optional parameters fit_params and returns a transformed version of X.

I think we should always use the same x-axis (i.e., `np.arange(0.5/100, 1, 1/100)`) for Platt Scaling and Isotonic Regression. `sorted_train_CX` has `num_item` of the base dataset in it, usually at the level of tens of thousands.

I have fixed it in the latest commit.

Reply:
I think we would only run .fit_transform() on the items in the training set, so not tens of thousands. But still, I get your point that we don't need all of the tiny jump points to make a good plot.

Actually, I was just thinking that the IsotonicRegresion instance, after we call `.fit()`, has a built-in `.predict()` method.

Why not store that instance inside our IsotonicRegressionCalibrationCurve instance, and use it to implement the `.get_calibrated_prob()` method that's needed in that subclass? Likely to be more efficient and less bug-prone than  what I implemented.

For plotting, you could do as you are suggesting: generate the 100 x-axis values like you do for Platt Scaling and pass them through `.get_calibrated_prob()` to get corresponding y-axis values.

Reply:
> I was just thinking that the IsotonicRegresion instance, after we call .fit(), has a built-in .predict() method.

This is indeed what I implemented yesterday. I didn't store the instance. Instead, I store the `x_axis` and corresponding `y_axis` for the 100 bins. The reason is that for any of `nonparametric binning, piecewise linear, adjusted piecewise linear`, they don't have a built-in instance. They are defined by a `x_axis` and a `y_axis`. I want to keep it consistent.

Of course I could change the stored object for `platt scaling, isotonic regression` to the fitted instance. The `.get_calibrated_prob()` method will become a wrapper of `.predict()` or `.predict_proba()`.

What do you think? Having a universal `x_axis` and `y_axis` design for all calibration curves or `fitted_instance` for `platt scaling, isotonic regression` and `x_axis` and `y_axis` for `nonparametric binning, piecewise linear, adjusted piecewise linear`?